letsencrypt/DOCS.md

```diff
@@ -320,6 +320,53 @@ dns:
 ```
 Use `ovh_endpoint: ovh-ca` for north America region.
 
+### RFC2136
+You will need to set up a server with RFC2136 (Dynamic Update) support with a TKEY (to authenticate the updates).  How to do this will vary depending on the DNS server software in use.  For Bind9, you first need to first generate an authenticate key by running
+```
+# dnssec-keygen -a HMAC-SHA512 -b 512 -n HOST letsencrypt
+Kletsencrypt.+165+20675
+```
+The key file (Kletsencrypt.+165+20675.key in this example) looks like the following:
+```
+# cat Kletsencrypt.+165+20675.key
+letsencrypt. IN KEY 512 3 165 Cj2SJThIYZqZO39HIOA8dYryzsLT3CI+m43m3yfGfTMvpyYw5DXjn5da hokrwyLe3MTboGkloKIsT6DUcTSdEA==
+```
+You don't need to publish this; just copy the key data into your named.conf file:
+```
+key "letsencrypt" {
+  algorithm hmac-sha512;
+  secret "Cj2SJThIYZqZO39HIOA8dYryzsLT3CI+m43m3yfGfTMvpyYw5DXjn5da hokrwyLe3MTboGkloKIsT6DUcTSdEA==";
+};
+```
+And ensure you have an update policy in place in the zone that uses this key to enable update of the correct domain
+```
+   update-policy {
+      grant letsencrypt name _acme-challenge.hassio.dom.ain. txt;
+   };
+```
+
+For this add-on you will need to supply all the rfc2136_* options.  Note that the rfc2136_port item is required (there is no default port in the add-on) and, most importantly, the the port number must be quoted because the add-on is expecting it to be a string and not a number.  Also be sure to copy in the key so certbot can authenticate to the DNS server.  Finally, the algorithm should be in all caps.
+
+An example configuration
+```yaml
+email: your-mail@dom.ain
+domains:
+  - hassio.dom.ain
+certfile: fullchain.pem
+keyfile: privkey.pem
+challenge: dns
+dns:
+  provider: dns-rfc2136
+  rfc2136_server: dns-server.dom.ain
+  rfc2136_port: '53'
+  rfc2136_name: letsencrypt
+  rfc2136_secret: >-
+    Cj2SJThIYZqZO39HIOA8dYryzsLT3CI+m43m3yfGfTMvpyYw5DXjn5da
+    hokrwyLe3MTboGkloKIsT6DUcTSdEA==
+  rfc2136_algorithm: HMAC-SHA512
+```
+
+One final note:  While using the FQDN has been tested and works, at least one user reports that using the FQDN for the server did not work for them.  It worked fine when they changed the configuration to use the server IP address.
```

Comment:
@ludeeus -- It is unclear whether this is actually a bug or a user configuration issue.  I agree we shouldn't document bugs, but one user did have an issue getting it to work and we (they) couldn't figure out why.  There was nothing obvious in the logs, which is why I think it was a user-configuration error.  My goal is documenting a "how to" to make this a one-stop for people to get this working for them, so that's why I added this text to help deal with a user error (which could be a server config issue, or something out of their control).

Instead of removing this text, do you have a suggestion for alternate text that would have helped this other user without appearing to document a bug?

Reply:
This sounds like a DNS bug/issue, until proven otherwise it should not be in here.

Reply:
If removing this will get you to merge, I'll take the suggestion.